heron/common/src/python/basics/event_looper.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2016 Twitter. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''event_looper.py'''
+import time
+import traceback
+import sys
+
+from abc import abstractmethod
+from heapq import heappush, heappop
+from heron.common.src.python.log import Log
+
+class EventLooper(object):
+  """Event Looper can block the thread on ``do_wait()`` and unblock on ``wake_up()``"""
+  def __init__(self):
+    self.should_exit = False
+    self.wakeup_tasks = []
+    self.timer_tasks = []
```

Comment:
This code isn't thread safe. Is it not intended to be invoked by multiple threads? If not it should be documented as such. If so, these objects should be accesses in a thread safe manner.